internal/parse/parse.go

```diff
@@ -37,6 +37,70 @@ var (
 	errBadVolSrc    = errors.New("must set volume source")
 )
 
+// BuildOutputOptions contains parsed field generated by parsing build output string
+type BuildOutputOption struct {
+	Path     string
+	IsDir    bool
+	IsStdout bool
+}
+
+// GetBuildOutput is responsible for parsing custom build output argument.
+// Takes `buildOutput` as string and returns path where output has to be generated returns BuildOutputOptions
+func GetBuildOutput(buildOutput string) (BuildOutputOption, error) {
+	isDir := true
+	isStdout := false
+	typeSelected := false
+	pathSelected := false
+	path := ""
+	if len(buildOutput) == 1 && buildOutput == "-" {
+		// Feature parity with buildkit, output tar to stdout
+		// Read more here: https://docs.docker.com/engine/reference/commandline/build/#custom-build-outputs
+		return BuildOutputOption{"", false, true}, nil
+	}
+	if strings.Contains(buildOutput, ",") {
+		tokens := strings.SplitN(buildOutput, ",", 2)
+		if len(tokens) != 2 {
+			return BuildOutputOption{path, isDir, isStdout}, fmt.Errorf("invalid build output options %q", buildOutput)
+		}
+		for _, option := range tokens {
+			arr := strings.SplitN(option, "=", 2)
+			if len(arr) != 2 {
+				return BuildOutputOption{path, isDir, isStdout}, fmt.Errorf("invalid build output options %q", buildOutput)
```

Comment:
I wouldn’t bother with filling in the `BuildOutputOption` values on error paths — the values are not expected to be used, and it only makes the `return`s look like success cases at a first glance.

Reply:
fixed